lib/iris/analysis/__init__.py

```diff
@@ -186,60 +186,7 @@ def matches_any(self, predicate):
         return any(self.matches(predicate))
 
 
-def coord_comparison(*cubes, object_get=None):
-    """
-    Convenience function to help compare coordinates on one or more cubes
-    by their metadata.
-
-    Return a dictionary where the key represents the statement,
-    "Given these cubes list the coordinates which,
-    when grouped by metadata, are/have..."
-
-    Keys:
-
-    * grouped_coords
-       A list of coordinate groups of all the coordinates grouped together
-       by their coordinate definition
-    * ungroupable
-       A list of coordinate groups which contain at least one None,
-       meaning not all Cubes provide an equivalent coordinate
-    * not_equal
-       A list of coordinate groups of which not all are equal
-       (superset of ungroupable)
-    * no_data_dimension
-       A list of coordinate groups of which all have no data dimensions on
-       their respective cubes
-    * scalar
-       A list of coordinate groups of which all have shape (1, )
-    * non_equal_data_dimension
-       A list of coordinate groups of which not all have the same
-       data dimension on their respective cubes
-    * non_equal_shape
-       A list of coordinate groups of which not all have the same shape
-    * equal_data_dimension
-       A list of coordinate groups of which all have the same data dimension
-       on their respective cubes
-    * equal
-       A list of coordinate groups of which all are equal
-    * ungroupable_and_dimensioned
-       A list of coordinate groups of which not all cubes had an equivalent
-       (in metadata) coordinate which also describe a data dimension
-    * dimensioned
-       A list of coordinate groups of which all describe a data dimension on
-       their respective cubes
-    * ignorable
-       A list of scalar, ungroupable non_equal coordinate groups
-    * resamplable
-        A list of equal, different data dimensioned coordinate groups
-    * transposable
-       A list of non equal, same data dimensioned, non scalar coordinate groups
-
-    Example usage::
-
-        result = coord_comparison(cube1, cube2)
-        print('All equal coordinates: ', result['equal'])
-
-    """
+def _dimensional_metadata_comparison(*cubes, object_get=None):
```

Comment:
Just add a comment or docstring to explain 
  * this is a 'generalised form' of the coord_comparison function
  * see there for a detailed behaviour description (especially, what it returns)
  * the 'object_get' is a fetch-all function for cube _DimensionalMetadata contents, like `Cube.coords()` (the default.

Reply:
@stephenworsley I don't quite understand the reason for making this private?
Wouldn't there be general utility in making this public and exposing this behaviour?

Reply:
@bjlittle My thinking behind making it private is that this should always serve as the most generic version of comparison for dimensional metadata. If it needs to change in the future to make it more generic, there is room to do so. To expose the new comparison behaviour, would it make sense to create specific public functions `cell_measure_comparison` and `ancillary_variable_comparison` instead?

Reply:
@bjlittle With that said, much of the output form `_dimensional_metadata_comparison` reads as coordinate specific. If this function were to be made public somehow, I think this would need to be addressed.

Reply:
Personally, I'm not convinced that we need public functionality for this.
If we _do_ want that, then I don't like ***either*** the existing form with the magic 'object_get' keyword, ***or*** the prospect of additional specific routines for `cell_measure_comparison` and `ancillary_variable_comparison`.  
Instead, I would push for a general-purpose `dimensional_metadata_comparison` routine with a rather nicer like an 'element_type' switch.

But in the end, I think we can just not do that, for now.